File: camerax_ios/ios/Classes/core/ZoomStateImpl.swift
//
//  ZoomStateImpl.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/16.
//

import Foundation

class ZoomStateImpl: PigeonApiDelegateZoomStateApi {
    func minZoomRatio(pigeonApi: PigeonApiZoomStateApi, pigeonInstance: ZoomState) throws -> Double {
        return pigeonInstance.minZoomRatio
    }
    
    func maxZoomRatio(pigeonApi: PigeonApiZoomStateApi, pigeonInstance: ZoomState) throws -> Double {
        return pigeonInstance.maxZoomRatio
    }
    
    func zoomRatio(pigeonApi: PigeonApiZoomStateApi, pigeonInstance: ZoomState) throws -> Double {
        return pigeonInstance.zoomRatio
    }
    
    func linearZoom(pigeonApi: PigeonApiZoomStateApi, pigeonInstance: ZoomState) throws -> Double {
        return pigeonInstance.linearZoom
    }
}

File: camerax_android/android/src/main/kotlin/dev/zeekr/camerax_android/core/ZoomStateImpl.kt
package dev.zeekr.camerax_android.core

import androidx.camera.core.ZoomState
import dev.zeekr.camerax_android.CameraXImpl
import dev.zeekr.camerax_android.PigeonApiZoomStateApi

class ZoomStateImpl(impl: CameraXImpl) : PigeonApiZoomStateApi(impl) {
    override fun minZoomRatio(pigeon_instance: ZoomState): Double {
        return pigeon_instance.minZoomRatio.toDouble()
    }

    override fun maxZoomRatio(pigeon_instance: ZoomState): Double {
        return pigeon_instance.maxZoomRatio.toDouble()
    }

    override fun zoomRatio(pigeon_instance: ZoomState): Double {
        return pigeon_instance.zoomRatio.toDouble()
    }

    override fun linearZoom(pigeon_instance: ZoomState): Double {
        return pigeon_instance.linearZoom.toDouble()
    }
}